callautomation-openai-sample-csharp/CallAutomationOpenAI/Program.cs

```diff
@@ -18,8 +19,39 @@
 //Grab the Cognitive Services endpoint from appsettings.json
 var cognitiveServicesEndpoint = builder.Configuration.GetValue<string>("CognitiveServiceEndpoint");
 
+string answerPromptSystemTemplate = """ 
+    You are an assisant designed to answer the customer query and analyze the sentiment score from the customer tone. 
+    You also need to determine the intent of the customer query and classify it into categories such as sales, marketing, shopping, etc.
+    Use a scale of 1-10 (10 being highest) to rate the sentiment score. 
+    Use the below format, replacing the text in brackets with the result. Do not include the brackets in the output: 
+    Content:[Answer the customer query briefly and clearly in two lines and ask if there is anything else you can help with] 
+    Score:[Sentiment score of the customer tone] 
+    Intent:[Determine the intent of the customer query] 
+    Category:[Classify the intent into one of the categories]
+    """;
+
+string helloPrompt = "Hello, thank you for calling! How can I help you today?";
+string timeoutSilencePrompt = "I�m sorry, I didn�t hear anything. If you need assistance please let me know how I can help you.";
+string goodbyePrompt = "Thank you for calling! I hope I was able to assist you. Have a great day!";
+string connectAgentPrompt = "I'm sorry, I was not able to assist you with your request. Let me transfer you to an agent who can help you further. Please hold the line and I'll connect you shortly.";
+string callTransferFailurePrompt = "It looks like all I can�t connect you to an agent right now, but we will get the next available agent to call you back as soon as possible.";
+string agentPhoneNumberEmptyPrompt = "I�m sorry, we're currently experiencing high call volumes and all of our agents are currently busy. Our next available agent will call you back as soon as possible.";
+string EndCallPhraseToConnectAgent = "Sure, please stay on the line. I�m going to transfer you to an agent.";
+
+string transferFailedContext = "TransferFailed";
+string connectAgentContext = "ConnectAgent";
+
+string agentPhonenumber = builder.Configuration.GetValue<string>("AgentPhoneNumber");
+string chatResponseExtractPattern = @"\s*Content:(.*)\s*Score:(.*\d+)\s*Intent:(.*)\s*Category:(.*)";
+
+var key = builder.Configuration.GetValue<string>("AzureOpenAIServiceKey");
+var endpoint = builder.Configuration.GetValue<string>("AzureOpenAIServiceEndpoint");
+
+var ai_client = new OpenAIClient(new Uri(endpoint), new AzureKeyCredential(key));
+
 //Register and make CallAutomationClient accessible via dependency injection
 builder.Services.AddSingleton(client);
+builder.Services.AddSingleton(ai_client);

```

Comment:
not able to comment below, but on line number 87, lets make it more robust in callbackurl construction